app/boards/arm/corneish_zen/corneish_zen_v1_right.dts

```diff
@@ -0,0 +1,96 @@
+/*
+*
+* Copyright (c) 2021 Darryl deHaan
+* SPDX-License-Identifier: MIT
+*
+*/
+
+#include "corneish_zen.dtsi"
+
+/{
+	chosen {
+		zephyr,display = &epd;
+		zmk,battery = &fuelgauge;
+	};
+
+	kscan0: kscan {
+		compatible = "zmk,kscan-gpio-matrix";
+		label = "KSCAN";
+
+		diode-direction = "col2row";
+		row-gpios
+			= <&gpio0 22 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+			, <&gpio1  0 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+			, <&gpio1  3 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+			, <&gpio1  1 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+			;
+
+		col-gpios
+			= <&gpio0 19 GPIO_ACTIVE_HIGH>
+			, <&gpio0 21 GPIO_ACTIVE_HIGH>
+			, <&gpio0 23 GPIO_ACTIVE_HIGH>
+			, <&gpio0 12 GPIO_ACTIVE_HIGH>
+			, <&gpio1  9 GPIO_ACTIVE_HIGH>
+			, <&gpio0  7 GPIO_ACTIVE_HIGH>
+			;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		blue_led: led_0 {
+			gpios = <&gpio0 26 GPIO_ACTIVE_HIGH>;
+			label = "Blue LED";
+		};
+	};
+};
+
+&default_transform {
+	col-offset = <6>;
+};
+
+&five_column_transform {
+	col-offset = <6>;
+};
+
+&i2c0 {
+	status = "ok";
+	compatible = "nordic,nrf-twim";
+	sda-pin = <16>;
+	scl-pin = <13>;
+	clock-frequency = <100000>; 
+
+	fuelgauge: bq274xx@55 {
```

Comment:
I thought this fuel gauge sensor driver didn't like the way our battery code probes for the value?

Reply:
To the best of my understanding the sensor exposes additional channels that could be read by the ZMK `battery.c` code, as added in #897. However just fetching the battery level channel (which is the only one utilized) seems to work OK without that PR and the current `battery.c` code. I have been testing this for 5-6 weeks and the reported battery levels have been pretty sensible.

Reply:
Thanks. Makes sense.